sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1931,8 +1931,7 @@ pi_result piQueueRetain(pi_queue Queue) {
 
 pi_result piQueueRelease(pi_queue Queue) {
   assert(Queue);
-  // Lock automatically releases when this goes out of scope.
-  std::lock_guard<std::mutex> lock(Queue->PiQueueMutex);
+  Queue->PiQueueMutex.lock();
```

Comment:
This change was necessary because the previous code will cause accessing Queue->PiQueueMutex at the end of this function (out of scope).
But, I added "delete Queue" at line 1954, so lock_guard deallocation using Queue->PiQueueMutex causes accessing memory that is already removed.

Reply:
there should be a way to still use scoped lock, please re-work

Reply:
If you look at the std::lock_guard, it uses (Queue->PiQueueMutex), and it needs Queue->PiQueueMutex again when it goes out of scope.
But, we already deallocated Queue by the time the control goes out of the scope, so it accesses the memory that is already removed.
The only way I can fix this ordering issue is to call std::mutex.lock() before changing the ref count, and std::mutex.unlock() after "delete Queue".
If you can think of how you can still use the scoped lock that is already deallocated, please let me know.